src/Slices/DuplicateInstance/UI/DuplicateInstancePage.test.tsx

```diff
@@ -0,0 +1,481 @@
+  QueryResolverImpl,
+  ServiceInstanceQueryManager,
+  ServiceInstanceStateHelper,
+} from "@/Data";
+import {
+  DeferredApiHelper,
+  dependencies,
+  DynamicQueryManagerResolverImpl,
+  MockEnvironmentModifier,
+  Service,
+  ServiceInstance,
+  StaticScheduler,
+} from "@/Test";
+import { words } from "@/UI";
+import { DependencyProvider } from "@/UI/Dependency";
+import { DuplicateInstancePage } from "./DuplicateInstancePage";
+
+function setup(entity = "a") {
+  const store = getStoreInstance();
+  const scheduler = new StaticScheduler();
+  const apiHelper = new DeferredApiHelper();
+  const authHelper = new KeycloakAuthHelper();
+  const queryResolver = new QueryResolverImpl(
+    new DynamicQueryManagerResolverImpl([
+      ServiceInstanceQueryManager(
+        apiHelper,
+        ServiceInstanceStateHelper(store),
+        scheduler,
+      ),
+    ]),
+  );
+
+  const commandResolver = new CommandResolverImpl(
+    new CommandManagerResolverImpl(store, apiHelper, authHelper),
+  );
+
+  const component = (
+    <MemoryRouter>
+      <DependencyProvider
+        dependencies={{
+          ...dependencies,
+          queryResolver,
+          commandResolver,
+          environmentModifier: new MockEnvironmentModifier(),
+        }}
+      >
+        <StoreProvider store={store}>
+          <DuplicateInstancePage
+            serviceEntity={Service[entity]}
+            instanceId={"4a4a6d14-8cd0-4a16-bc38-4b768eb004e3"}
+          />
+        </StoreProvider>
+      </DependencyProvider>
+    </MemoryRouter>
+  );
+
+  return { component, apiHelper, scheduler };
+}
+
+test("Duplicate Instance View shows failed state", async () => {
+  const { component, apiHelper } = setup();
+  render(component);
+
+  expect(
+    await screen.findByRole("generic", { name: "DuplicateInstance-Loading" }),
+  ).toBeInTheDocument();
+
+  apiHelper.resolve(Either.left("error"));
+
+  expect(
+    await screen.findByRole("generic", { name: "DuplicateInstance-Failed" }),
+  ).toBeInTheDocument();
+});
+
+test("DuplicateInstance View shows success form", async () => {
+  const { component, apiHelper } = setup();
+  render(component);
+  const { service_entity } = ServiceInstance.a;
+
+  expect(
+    await screen.findByRole("generic", { name: "DuplicateInstance-Loading" }),
+  ).toBeInTheDocument();
+
+  apiHelper.resolve(Either.right({ data: ServiceInstance.a }));
+
+  expect(
+    await screen.findByRole("generic", { name: "DuplicateInstance-Success" }),
+  ).toBeInTheDocument();
+
+  const bandwidthField = screen.getByText("bandwidth");
+  expect(bandwidthField).toBeVisible();
+
+  await act(async () => {
+    await userEvent.type(bandwidthField, "3");
+  });
+  await act(async () => {
+    await userEvent.click(screen.getByText(words("confirm")));
+  });
+
+  expect(apiHelper.pendingRequests).toHaveLength(1);
+
+  expect(apiHelper.pendingRequests[0]).toEqual({
+    method: "POST",
+    url: `/lsm/v1/service_inventory/${service_entity}`,
+    body: {
```

Comment:
it seems like body of the request is the same here and in the line 214 so we could extract it, but it's hard to compare those from this view so I could me mistaken

Reply:
they are almost the same, in the second one a value in the embedded entity has been changed.